### PR TITLE
fix: mine depend txs in one block

### PR DESCRIPTION
**Pull request by jjyr**

### CODE REVIEW

File: miner/src/block_assembler.rs [hunk 1]
```diff
@@ -324,8 +374,13 @@ impl<CS: ChainStore + 'static> BlockAssembler<CS> {
         // bytes, they really serve the same purpose at the moment
         let block_reward = self.shared.block_reward(header.number() + 1);
         let mut fee = 0;
-        for pe in pes {
-            fee += self.shared.calculate_transaction_fee(&pe.transaction)?;
+        for (i, pe) in pes.iter().enumerate() {
+            // depends cells may produced from previous tx
+            let fee_calculator = FeeCalculator {
```
Comment (quake) [inline miner/src/block_assembler.rs]: It's better to use `HashMap` in FeeCalculator as a previously calculated txs capacity cache, and change the `calculate_transaction_fee` function to mut, which will simplify code here.

File: test/src/specs/transaction/depend_tx_in_same_block.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
+use crate::{sleep, Net, Spec};
+use log::info;
+
+pub struct DepentTxInSameBlock;
```
Comment (quake) [inline test/src/specs/transaction/depend_tx_in_same_block.rs]: Please move this spec to specs/mining.rs

Comment (jjyr) [reply test/src/specs/transaction/depend_tx_in_same_block.rs]: I'm not sure, even this bug caused by miner, in general, this is a transaction/block verifier related spec.

Comment (quake) [reply test/src/specs/transaction/depend_tx_in_same_block.rs]: Verifier uses different logic to verify cellbase capacity, this PR only fix the bug in miner's `create_cellbase_transaction` function, so it's better to move this spec to `mining.rs`

**Pull request merged**